### PR TITLE
RUST-1731 Remove async-std support and make tokio an unconditional dependency

### PR DESCRIPTION
**Pull request by abr-egn**
RUST-1731

A lot of the `runtime` module could just go away with this; the few things remaining are actually doing something useful.

### CODE REVIEW

File: src/sdam/description/topology/server_selection/test/logic.rs [hunk 1]
```diff
@@ -103,60 +103,54 @@ async fn run_test(test_file: TestFile) {
             );
         }
     } else if test_file.error == Some(true) {
-        // skip on sync to avoid compilation conflicts with the sync version of
-        // ClientOptions::parse.
-        #[cfg(not(any(feature = "sync", feature = "tokio-sync")))]
-        {
-            use crate::{
```
Comment (abr-egn) [inline src/sdam/description/topology/server_selection/test/logic.rs:110]: Github's diff is mismatching here; the only change is removing the `#[cfg]`.

**Review by isabelatkinson — COMMENTED** [review]
very exciting! just a few small comments/suggestions

File: src/runtime/stream.rs [hunk 2]
```diff
@@ -78,45 +78,23 @@ mod unix {
     #[derive(Debug)]
     pub(crate) enum AsyncUnixStream {
```
[the remark this answers is not in the document]
Comment (abr-egn) [reply src/runtime/stream.rs]: Good call - removed the layers of wrapping here, just kept the toplevel enum to distinguish tcp/tls/unix.

**Review by isabelatkinson — APPROVED** [review]
> +774 −4,897

awesome! lgtm!

**Pull request merged**